src/scala/microsoft-spark-2-3/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -43,7 +44,12 @@ object DotnetRunner extends Logging {
       throw new IllegalArgumentException("At least one argument is expected.")
     }
 
-    validateSparkVersions
+    DotnetUtils.validateSparkVersions(
+      Option(SparkEnv.get).map(_.conf).getOrElse(new SparkConf()),
```

Comment:
question. this SparkEnv will be set for `spark-submit` and livy submission right?

Reply:
I know it works for `spark-submit`, not sure about livy submission.

Reply:
Hmm, to me `SparkEnv.get` should always return None since it's set in during SparkContext initialization? I think the reason this works is because of
https://github.com/apache/spark/blob/06d40696dc27c39824666a70e7c673880959a681/core/src/main/scala/org/apache/spark/SparkConf.scala#L70

And
https://github.com/apache/spark/blob/06d40696dc27c39824666a70e7c673880959a681/core/src/main/scala/org/apache/spark/deploy/SparkApplication.scala#L47

Were you getting `Some` from `SparkEnv.get`?

Reply:
`SparkEnv.get` is `None`.  We can pass in a `new SparkConf` directly or just get the values from `sys.props` instead if you prefer.

Reply:
If it's always None, I would remove it since it just adds confusion and document clearly. For example:
```scala
// Load defaults set in the system properties when this application is launched:
// https://github.com/apache/spark/blob/06d40696dc27c39824666a70e7c673880959a681/core/src/main/scala/org/apache/spark/deploy/SparkApplication.scala#L47
val sparkConf = new SparkConf(loadDefaults = true)
```

Reply:
I opted to go with the `sys.props` since [RRunner.scala](https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/core/src/main/scala/org/apache/spark/deploy/RRunner.scala#L48) is also getting the conf this way.

Reply:
OK, either one is fine as long as `SparkEnv.get` is removed.